brave/src/main/java/brave/propagation/TraceContext.java

```diff
@@ -445,21 +445,30 @@ boolean invalidIdLength(Object key, int length, int max) {
       return true;
     }
 
-    boolean isNull(Object key, String maybeNull) {
+    static boolean isNull(Object key, String maybeNull) {
       if (maybeNull != null) return false;
       Platform.get().log("{0} was null", key, null);
       return true;
     }
 
-    void maybeLogNotLowerHex(String notLowerHex) {
+    /** Helps differentiate a parse failure from a successful parse of all zeros. */
+    static boolean isAllZeros(String value, int beginIndex, int endIndex) {
```

Comment:
later, we could consider sneaky sharing between b3 single format and here, as the parsing requirements are similar

Reply:
I definitely want to refactor this afterwards as it is annoying. for now, just fixing the bug